### PR TITLE
Add planner reinitialization for elasticity 

### PR DESCRIPTION
**Pull request by zekemorton**
When either growing or shrinking the resource graph, there is no means of reinitializing the planner with the proper resource counts. If the planner is not reinitialized, the original totals from when the graph was created will remain. If it is initialized again, it will loose all allocated counts. This PR combined with the effort from converting planner_multi into a boost multi index address this issue. 

This PR adds a check in the initialization function to see if a planner already exists, and if it does, it updates the total counts rather than creating a new planner without the allocated counts.

### CODE REVIEW

File: resource/traversers/dfu_impl.cpp [hunk 1]
```diff
@@ -1131,7 +1131,12 @@ int dfu_impl_t::prime_pruning_filter (const subsystem_t &s, vtx_t u,
         }
```
Comment (milroy) [inline resource/traversers/dfu_impl.cpp:1131]: Nit: commit should wrap at 72 characters per RFC 1: https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_1.html#patch-requirements

File: resource/traversers/dfu_impl.cpp [hunk 2]
```diff
@@ -1140,6 +1145,9 @@ int dfu_impl_t::prime_pruning_filter (const subsystem_t &s, vtx_t u,
             goto done;
         }
         (*m_graph)[u].idata.subplans[s] = p;
+    } else {
+        planner_multi_update ((*m_graph)[u].idata.subplans[s], &avail [0], 
+                               &types [0], types.size ());
```
Comment (milroy) [inline resource/traversers/dfu_impl.cpp:1150]: Nit: spacing.
```suggestion
        planner_multi_update ((*m_graph)[u].idata.subplans[s], &avail[0], 
                               &types[0], types.size ());
```

File: t/data/resource/commands/elastic/cmds09.in [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+match allocate @TEST_SRCDIR@/data/resource/jobspecs/elastic/test-ma-node.yaml
+match allocate_orelse_reserve @TEST_SRCDIR@/data/resource/jobspecs/elastic/test-ma-node.yaml
+find status=up
+find sched-now=allocated
+find sched-future=reserved
+match allocate @TEST_SRCDIR@/data/resource/jobspecs/elastic/test-ma-node.yaml
+attach @TEST_SRCDIR@/data/resource/jgfs/elastic/node-add-test.json
+find status=up
+find sched-now=allocated
+find sched-future=reserved
```
Comment (milroy) [inline t/data/resource/commands/elastic/cmds09.in:11]: I'd add `find sched-now=free` here as well. After that, please add a match allocate request for the newly attached resources and then check the allocation state again.

**Review by milroy — CHANGES_REQUESTED** [review]
The PR looks good. I found a couple of nits and identified a few more commands to run in the testsuite.

After you make those changes this PR will be good to merge.

**Review by milroy — DISMISSED** [review]
Thanks for the quick turnaround. LGTM!

After rebasing on the current `master` state go ahead and set MWP.

**Comment by zekemorton** [comment]
@milroy looks like it wants another approval after rebasing

**Review by milroy — APPROVED** [review]
Double approved!

**Pull request merged**